app/controllers/api/v1/wallets_controller.rb

```diff
@@ -11,14 +11,42 @@ def index
 
   # POST /api/v1/wallets
   def create
-    @wallet = WalletCreator.new(account: account).call(wallet_params, tokens_to_provision: tokens_to_provision)
+    ActiveRecord::Base.transaction do
+      @wallets = WalletCreator.new(account: account).call(wallets_params)
 
-    if @wallet.persisted?
-      render 'show.json', status: :created
-    else
-      @errors = @wallet.errors
+      if @wallets.all?(&:persisted?)
+        render 'index.json', status: :created
+      else
+        @errors = {}
 
-      render 'api/v1/error.json', status: :bad_request
+        @wallets.each_with_index.map do |wallet, i|
```

Comment:
```suggestion
         @wallets.map.with_index do |wallet, i|
```
It just reads a little bit better